### PR TITLE
CLOUDP-57864: Backup: Start a restore job

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ [CLOUDP-57864](https://jira.mongodb.org/browse/CLOUDP-57864)

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/cli/atlas_backups_restores_start.go [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+func (opts *atlasBackupsRestoresStartOpts) isAutomatedRestore() bool {
+	return opts.method == automatedRestore
+}
+
+func (opts *atlasBackupsRestoresStartOpts) isHTTP() bool {
+	return opts.method == httpRestore
+}
+
+func (opts *atlasBackupsRestoresStartOpts) validateParams() error {
+	if opts.clusterName == "" && opts.clusterID == "" {
```
Comment (andreaangiolillo) [inline internal/cli/atlas_backups_restores_start.go]: I am wondering if we should use a switch here

Comment (gssbzn) [reply internal/cli/atlas_backups_restores_start.go]: this definitely needs more love, I'll see what I can do today with fresh eyes

**Comment by gssbzn** [comment]
@andreaangiolillo this should be RFAL

File: internal/cli/atlas_backups_restores_start.go [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+	}
+	cmd := &cobra.Command{
+		Use:       "start",
+		Short:     "Start a restore job.",
+		Args:      cobra.ExactValidArgs(1),
+		ValidArgs: []string{automatedRestore, httpRestore},
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if opts.isAutomatedRestore() {
+				_ = cmd.MarkFlagRequired(flags.TargetProjectID)
+
```
Comment (andreaangiolillo) [inline internal/cli/atlas_backups_restores_start.go]: [np] this could be a function

File: internal/cli/atlas_backups_restores_start.go [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+				if config.Service() == config.CloudService {
+					_ = cmd.MarkFlagRequired(flags.ClusterName)
+				} else {
+					_ = cmd.MarkFlagRequired(flags.ClusterID)
+				}
+			}
```
Comment (andreaangiolillo) [inline internal/cli/atlas_backups_restores_start.go]: [np] this could be a function

Comment (gssbzn) [reply internal/cli/atlas_backups_restores_start.go]: updated

**Pull request merged**